### PR TITLE
Implement Redis-Go on Public API

### PR DESCRIPTION
**Pull request by rachadiannovansyah**
Overview
Implement Redis on Public API

- create own Middleware to check `verify cache`
- set it to public API (slug news api)
- change attribute type of `NullString`, `NullTime`, `NullInt` instead of Pointer

cc: https://github.com/orgs/jabardigitalservice/teams/jds-backend

Evidence
project: Portal Jabar
title: Redis cache on public API to enhance performance
participants: @rachadiannovansyah @ayocodingit

### CODE REVIEW

File: core-service/src/modules/news/delivery/http/public_news_handler.go [hunk 1]
```diff
@@ -83,6 +88,13 @@ func (h *PublicNewsHandler) GetBySlug(c echo.Context) error {
 	newsRes := domain.DetailNewsResponse{}
 	copier.Copy(&newsRes, &news)
 
+	// set ttl cache 5 minutes after store in redis
+	ttl := 5 * time.Minute
```
Comment (ayocodingit) [inline core-service/src/modules/news/delivery/http/public_news_handler.go:92]: Better Sebaiknya di simpan ke env van 😁  biar dinamis

File: core-service/src/middleware/permission.go [hunk 2]
```diff
@@ -35,3 +36,23 @@ func CheckPermission(permission string) echo.MiddlewareFunc {
...
 	}
 }
+
+func CheckCache() echo.MiddlewareFunc {
+	return func(next echo.HandlerFunc) echo.HandlerFunc {
+		return func(c echo.Context) error {
+			memcached := helpers.GetCachedData(c)
+
+			if memcached != "" {
+				var cacheRes domain.DetailNewsResponse
```
Comment (ayocodingit) [inline core-service/src/middleware/permission.go:46]: Btw ini domainnya hanya untuk detail news yah gak dibuat di buat lebih global lagi van?

**Review by ayocodingit — APPROVED** [review]

File: core-service/src/modules/news/delivery/http/public_news_handler.go (hunk 1 above)
Comment (rachadiannovansyah) [reply core-service/src/modules/news/delivery/http/public_news_handler.go]: yes looks better, here we go man https://github.com/jabardigitalservice/portal-jabar-services/pull/253/commits/f47d73aff16007b0d8a39ffecffeb1e36cc2d2e5

File: core-service/src/middleware/permission.go (hunk 2 above)
Comment (rachadiannovansyah) [reply core-service/src/middleware/permission.go:46]: yes so far this response has been used (res of detail news response)

**Review by ayocodingit — APPROVED** [review]

**Pull request merged**